dbos/_core.py

```diff
@@ -213,10 +223,11 @@ def persist(func: Callable[[], R]) -> R:
             output = func()
             status["status"] = "SUCCESS"
             status["output"] = _serialization.serialize(output)
-            if status["queue_name"] is not None:
-                queue = dbos._registry.queue_info_map[status["queue_name"]]
-                dbos._sys_db.remove_from_queue(status["workflow_uuid"], queue)
-            dbos._sys_db.buffer_workflow_status(status)
+            if not dbos.debug_mode:
```

Comment:
I think we need the same `if not dbos.debug_mode` logic for exception handling, so we don't try to update workflow status during debug mode.

When I debugged a workflow that threw an exception, I got `Exception: called update_workflow_status in debug mode`